baseplate/context/memcache/lib.py

```diff
@@ -0,0 +1,233 @@
+        :returns: value serialized as str, flags int.
+
+        """
+
+        if isinstance(value, string_types):
+            serialized = value
+            flags = 0
+        elif isinstance(value, int):
+            serialized = "%d" % value
+            flags = Flags.INTEGER
+        elif isinstance(value, long):
+            serialized = "%d" % value
+            flags = Flags.LONG
+        else:
+            serialized = json.dumps(value)
+            flags = Flags.JSON
+
+        if (compress_level and
+                min_compress_length and
+                len(serialized) > min_compress_length):
+            compressed = zlib.compress(serialized, compress_level)
+            flags |= Flags.ZLIB
+            return compressed, flags
+        else:
+            return serialized, flags
+
+    return dump_and_compress
+
+
+class PickleFlags(object):
+    """Memcached client flags
+
+    Flags are an arbitrary 16-bit unsigned integer that the memcache server
+    stores along with the data and sends back when the item is retrieved.
+    Clients may use this as a bit field to store data-specific information;
+    this field is opaque to the server.
+
+    """
+
+    PICKLE = 1 << 0
+    INTEGER = 1 << 1
+    LONG = 1 << 2
+    ZLIB = 1 << 3
+
+
+def decompress_and_unpickle(key, serialized, flags):
+    """Deserialization method compatible with pickle_and_compress().
+
+    :param str key: the memcached key.
+    :param str serialized: the serialized object returned from memcached.
+    :param int flags: value stored and returned from memcached for the client
+        to use to indicate how the value was serialized.
+    :returns str value: the deserialized value.
+
+    """
+
+    if flags & PickleFlags.ZLIB:
+        serialized = zlib.decompress(serialized)
+        flags ^= PickleFlags.ZLIB
+
+    if flags == 0:
+        return serialized
+    elif flags == PickleFlags.INTEGER:
+        # python3 doesn't have a long integer type, so all integers are written
+        # with PickleFlags.INTEGER. This means that a value written by a python3
+        # client can exceed the maximum integer value (sys.maxint). This
+        # appears to be ok--python2 will automatically convert to long if the
+        # value is too large.
+        return int(serialized)
+    elif flags == PickleFlags.LONG:
+        return long(serialized)
+    elif flags == PickleFlags.PICKLE:
+        try:
+            return pickle.loads(serialized)
+        except Exception:
+            logging.info('Pickle error', exc_info=True)
+            return None
+    else:
+        logging.info('unrecognized flags')
+        return serialized
+
+
+def make_pickle_and_compress_fn(min_compress_length=0, compress_level=1):
+    """Create a serialization method compatible with decompress_and_unpickle().
+
+    The resulting method is a chain of pickling and zlib compression.
+
+    This serializer is compatible with pylibmc.
+
```

Comment:
can you add a `.. warning:: Please don't friggin use this` kind of thing here too please?